File: apache2/www/change_password.php
<?php

// breaks JSON error output!
//error_reporting(E_ALL);
//ini_set('display_errors', '1');

function send_error($message)
{
  $jTableResult = array();
  $jTableResult['Result'] = "ERROR";
  $jTableResult['Message'] = $message;
  print json_encode($jTableResult);
}

// Create connection
$dbconn = new mysqli('localhost', 'potsbliz', 'potsbliz', 'potsbliz');
if ($dbconn->connect_error) {
  send_error($dbconn->connect_error);
} 
else
{
	$sql = "UPDATE mysql_auth SET passwd=MD5('" . $_POST["newpw"] . "') WHERE username='admin' AND passwd=MD5('" . $_POST["oldpw"] . "');";
	if ($dbconn->query($sql))
	{
	  if ($dbconn->affected_rows == 1)
	  {
	    //Return result to jTable
	    $jTableResult = array();
	    $jTableResult['Result'] = "OK";
	    $jTableResult['Message'] = "Password successfully changed";
	    print json_encode($jTableResult);
	  }
	  else
	  {
	    send_error("Old password is wrong!");
	  }
	}
	else
	{
	  send_error($dbconn->error);
	}
}
$dbconn->close();

?>

File: apache2/www/change_password.py
# POTSBLIZ - Plain Old Telephone Service Beyond Local IP Stack
# (C)2015  - Norbert Huffschmid - GNU GPL V3 

import json
import potsbliz.config as config

def index(req):
    req.content_type = 'application/json; charset=UTF8'
    
    try:
        config.update_password(req.form.getfirst('oldpw'), req.form.getfirst('newpw'))
        return json.dumps({ 'Result': 'OK', 'Message': 'Password successfully changed' })
    except Exception, e:
        return json.dumps({ 'Result': 'ERROR', 'Message': str(e) })
